subgraph/src/entities/PeriodIndexCounter.ts

```diff
@@ -0,0 +1,14 @@
+import { PeriodIndexCounter } from "../../generated/schema";
+import { BigInt } from "@graphprotocol/graph-ts";
+
+export function getAndIncrementCounter(id: string): BigInt {
```

Comment:
Could we change the function name to `getAndIncrementPeriodCounter` or `getAndIncrementPeriodIndexCounter` mostly to avoid confusion with the `Counter` entity.

Reply:
I've updated this to getAndIncrementPeriodCounter